core/src/query.rs

```diff
@@ -123,24 +127,26 @@ impl MongoStatement for MongoQuery {
             "statement": &self.query,
         }}];
 
-        let cursor: Cursor<Document> = match self.query_timeout {
-            Some(i) => {
-                if i > 0 {
-                    let opt = AggregateOptions::builder()
-                        .max_time(Duration::from_millis(i as u64))
-                        .build();
-                    db.aggregate(pipeline, opt)
-                        .map_err(Error::QueryExecutionFailed)?
-                } else {
-                    // If the query timeout is 0, it means "no timeout"
-                    db.aggregate(pipeline, None)
-                        .map_err(Error::QueryExecutionFailed)?
-                }
-            }
-            _ => db
-                .aggregate(pipeline, None)
-                .map_err(Error::QueryExecutionFailed)?,
+        let opt = AggregateOptions::builder().comment_bson(Some(stmt_id));
+        // If the query timeout is 0, it means "no timeout"
+        let options = if self.query_timeout.is_some_and(|timeout| timeout > 0) {
+            opt.max_time(Duration::from_millis(self.query_timeout.unwrap() as u64))
+                .build()
+        } else {
+            opt.build()
```

Comment:
Nice refactor here! I've never seen `is_some_and` used before lol.

Reply:
++ it came from @nbagnard 's suggestion, I think it came out very clean 😄